src/main/java/com/sender/email/EmailController.java

```diff
@@ -0,0 +1,53 @@
+package com.sender.email;
+
+import com.sender.email.models.Email;
+import com.sender.email.repos.EmailProcessing;
+import com.sender.email.service.EmailSender;
+import lombok.AllArgsConstructor;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/emails")
+@AllArgsConstructor
+public class EmailController {
+    EmailProcessing processing;
+    EmailSender emailSender;
+
+    @GetMapping
+    public List<Email> getEmails() {
+        return processing.getAll();
+    }
+
+    @PostMapping(consumes = "application/json")
+    public void newEmail(@RequestBody List<Email> email){
+        processing.addNewEmail(email);
+        emailSender.sendAll();
+    }
+
+    @PutMapping(value = "/{id}/date", produces = "application/json")
+    public ResponseEntity<String> updateDate(@PathVariable(value="id") int id,@RequestParam(value="date") @DateTimeFormat(pattern = "dd-MM-yyyy HH:mm") Date newDate){
+        int last_id = processing.getAll().get(processing.getAll().size()-1).getId();
```

Comment:
_size - 1_ is always dangerous way to get data from array, especially without additional size verification.